hail/src/main/scala/is/hail/methods/LocalLDPrune.scala

```diff
@@ -357,7 +356,7 @@ case class LocalLDPrune(
       it.map { rv =>
         region.clear()
         rvb.set(region)
-        rvb.start(tableType.rowType.physicalType)
+        rvb.start(tableType.canonicalPType)
```

Comment:
should be bpvType I think?

Reply:
also the canonicalRVDType above should be RVDType(bpvType, Array("locus", "alleles"))

Reply:
I don't yet understand this section of code well, but making the following changes results in an error in the r2 test

(added ` rvb.addFields(bpvType, rv, fieldIndicesToAdd)` , ` rvdLP.mapPartitions(
      RVDType(bpvType, Array("locus", "alleles"))
    )`

```scala
 val fieldIndicesToAdd = Array("locus", "alleles", "mean", "centered_length_rec")
      .map(field => bpvType.fieldIdx(field))
    val sitesOnly = rvdLP.mapPartitions(
      RVDType(bpvType, Array("locus", "alleles"))
    )({ it =>
      val region = Region()
      val rvb = new RegionValueBuilder(region)
      val newRV = RegionValue(region)

      it.map { rv =>
        region.clear()
        rvb.set(region)
        rvb.start(bpvType)
        rvb.startStruct()
        rvb.addFields(bpvType, rv, fieldIndicesToAdd)
        rvb.endStruct()
        newRV.setOffset(rvb.end())
        newRV
      }
    })
```

```sh
hail.methods.LocalLDPruneSuite)

Gradle suite > Gradle test > is.hail.methods.LocalLDPruneSuite.testIsLocallyUncorrelated FAILED
    java.lang.IllegalArgumentException at LocalLDPruneSuite.scala:355
Running test: Test method testR2(is.hail.methods.LocalLDPruneSuite)
```

Tired, not thinking clearly

Reply:
Will finish digging through tomorrow:

```scala
java.lang.IllegalArgumentException: requirement failed
	at scala.Predef$.require(Predef.scala:212)
	at is.hail.expr.ir.TableValue.<init>(TableValue.scala:45)
	at is.hail.methods.LocalLDPrune.execute(LocalLDPrune.scala:368)
	at is.hail.methods.LocalLDPrune$$anonfun$apply$3.apply(LocalLDPrune.scala:286)
	at is.hail.methods.LocalLDPrune$$anonfun$apply$3.apply(LocalLDPrune.scala:284)
	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1.apply(ExecuteContext.scala:8)
	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1.apply(ExecuteContext.scala:7)
	at is.hail.utils.package$.using(package.scala:596)
	at is.hail.annotations.Region$.scoped(Region.scala:11)
	at is.hail.expr.ir.ExecuteContext$.scoped(ExecuteContext.scala:7)
	at is.hail.methods.LocalLDPrune$.apply(LocalLDPrune.scala:284)
	at is.hail.methods.LocalLDPruneSuite.testIsLocallyUncorrelated(LocalLDPruneSuite.scala:355)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccess
```